pkg/services/sqlstore/sqlstore_test.go

```diff
@@ -66,44 +68,50 @@ var sqlStoreTestCases = []sqlStoreTest{
 		dbHost:        "[::1]",
 		connStrValues: []string{"host=::1", "port=5432"},
 	},
+	{
+		name:  "Invalid database URL",
+		dbURL: "://invalid.com/",
+		err:   errors.New("parse \"://invalid.com/\": missing protocol scheme"),
+	},
 }
 
 func TestSQLConnectionString(t *testing.T) {
-	Convey("Testing SQL Connection Strings", t, func() {
-		t.Helper()
-
-		for _, testCase := range sqlStoreTestCases {
-			Convey(testCase.name, func() {
-				sqlstore := &SQLStore{}
-				sqlstore.Cfg = makeSQLStoreTestConfig(testCase.dbType, testCase.dbHost)
-				sqlstore.readConfig()
+	for _, testCase := range sqlStoreTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			sqlstore := &SQLStore{}
+			sqlstore.Cfg = makeSQLStoreTestConfig(t, testCase.dbType, testCase.dbHost, testCase.dbURL)
+			if err := sqlstore.readConfig(); err != nil {
+				require.Equal(t, testCase.err.Error(), err.Error())
+				return
+			}
```

Comment:
Some problems with this approach:

* If `testCase.err` is nil, you get a segfault
* If `testCase.err` is non-nil and `err` is nil, the former gets ignored